src/taskpane/server.js

```diff
@@ -192,26 +228,50 @@ function save() {
   }
 }
 
+//Function to check if a given sheet name exists or not in the workbook 
+async function checkSheetExists(sheetName) {
+  Excel.run(function (context) {
+    var sheets = context.workbook.worksheets;
+    sheets.load("items/name");
 
+    return context.sync()
+      .then(function () {
+        sheets.items.forEach(function (sheet) {
+          if (sheet.name == sheetName) {
+            return true;
+          }
+        });
+      });
+  });
+  return false;
+}
 
 //clears active sheet in spreadsheet
-function clearAll() {
+function clearAll(model) {
 
   if (IS_GOOGLE) {
     // get active spreadsheet
     var spreadSheet = SpreadsheetApp.getActiveSpreadsheet(),
       sheet = spreadSheet.getActiveSheet(),
-      lastColumn = sheet.getMaxColumns(),
-      lastRow = sheet.getMaxRows();
+      lastColumn = sheet.getLastColumn(),
+      lastRow = sheet.getLastRow();
+
+    //minimum requirements
+    if (lastColumn == 0) { lastColumn = 1 };
+    if (lastRow == 0) { lastRow = 1 };
 
     // Reset sheet name
     sheet.setName(new Date().getTime());
     sheet.clear();
 
     // clears data validations and notes from the entire sheet
-    var range = sheet.getRange(1, 1, lastRow, lastColumn);
+    var range = sheet.getRange(1, 1, sheet.getMaxRows() - 1, lastColumn);

```

Comment:
Basically, to allow users to add new data. getLastRow() brings only rows with some content on it (<>""). getMaxRows() brings the entire spreadsheet. It means I'm applying (or cleaning before applying) the styles to the whole sheet, instead of just a few lines. If the user wants to append more data, it will be ready for it. There was a bug here that we didn't catch before: if let's say, the user created a custom dropdown at a blank row after the dataset, when getting the data again, that custom dropdown would remain - possibly causing errors. This change makes sure to clean the whole spreadsheet before getting new data - that's safer.